src/common/snippets/src/lowered/pass/init_loops.cpp

```diff
@@ -38,13 +40,45 @@ int64_t get_output_stride(size_t dim, const VectorDims& shape) {
 
 InitLoops::InitLoops() : Pass() {}
 
-void InitLoops::init_is_incremented(const LinearIR::LoopManager::LoopInfoPtr& loop_info) {
+void InitLoops::init_is_incremented(const LinearIR::LoopManager::LoopInfoPtr& loop_info, size_t loop_id) {
     auto loop_entries = loop_info->get_entry_points();
     auto loop_exits = loop_info->get_exit_points();
-    auto update = [](std::vector<LoopPort>& ports) {
+    auto update = [=](std::vector<LoopPort>& ports) {
         for (auto& port : ports) {
-            if (!ov::is_type<op::MemoryAccess>(port.expr_port->get_expr()->get_node())) {
+            const auto& expr = port.expr_port->get_expr();
+            if (!std::dynamic_pointer_cast<modifier::MemoryAccess>((expr->get_node()))) {
                 port.is_incremented = false;
+            } else if (expr->get_loop_ids().back() != loop_id) {
+                // Note: LoopPort connected to Buffer between two loops should not be incremented in the outermost loop
+                // Consider the example below:
+                //     Store; Loop ids [1,2,3]
+                //     IntermediateMemoryBuffer; Loop ids [1]
+                //     Load; Loop ids [1, 4, 5]
+                // Store is exit port of Loop-1, but it should be incremented only in Loop-2 and Loop-3. Similar with Load.
+                auto is_ignored = [=](const ExpressionPtr& expr){
```

Comment:
```suggestion
                auto is_ignored = [=](const ExpressionPtr& expr) {
```
BTW that's strange that code style did not suggest this change. Does it even run on the lowered passes files?

Reply:
Yes, it's strange indeed. Usually it doesn't allow to skip spaces. Fixed.